usaspending_api/etl/elasticsearch_loader_helpers/controller.py

```diff
@@ -1,119 +1,145 @@
+        self.config["processes"] = min(self.config["processes"], self.config["partitions"])
 
-        if self.config["process_deletes"]:
-            process_list.append(
-                Process(
-                    name="S3 Deleted Records Scrapper Process",
-                    target=deleted_transactions if self.config["load_type"] == "transactions" else deleted_awards,
-                    args=(self.elasticsearch_client, self.config),
-                )
+        logger.info(
+            format_log(
+                f"Hailing {self.config['partitions']:,} heroes"
+                f" to handle {self.record_count:,} {self.config['data_type']} records"
+                f" in squads of {self.config['processes']:,}"
             )
-            process_list[-1].start()  # start S3 csv fetch proces
-            while process_list[-1].is_alive():
-                logger.info(format_log("Waiting to start ES ingest until S3 deletes are complete"))
-                sleep(7)  # add a brief pause to make sure the deletes are processed in ES
-
-        if updated_record_count != 0:
-            process_list[1].start()  # start ES ingest process
-
-        while True:
-            sleep(10)
-            if process_guarddog(process_list):
-                raise SystemExit("Fatal error: review logs to determine why process died.")
-            elif all([not x.is_alive() for x in process_list]):
-                logger.info(format_log("All ETL processes completed execution with no error codes"))
-                break
-
-    def create_download_jobs(self) -> Tuple[Queue, int]:
-        download_queue = Queue()
-        for job_number, fiscal_year in enumerate(self.config["fiscal_years"], start=1):
-            index = self.config["index_name"]
-            filename = str(self.config["directory"] / f"{fiscal_year}_{self.config['load_type']}.csv")
-
-            new_job = DataJob(job_number, index, fiscal_year, filename)
-
-            if Path(filename).exists():
-                Path(filename).unlink()
-            download_queue.put(new_job)
-        return download_queue, job_number
+        )
+
+        self.workers = [self.create_worker(j) for j in range(self.config["partitions"])]
+
+        if self.config["create_new_index"]:
+            # ensure template for index is present and the latest version
+            call_command("es_configure", "--template-only", f"--load-type={self.config['data_type']}s")
+            create_index(self.config["index_name"], instantiate_elasticsearch_client())
+
+    def launch_workers(self):
+        _abort = Event()  # Event which signals an error occured in a subprocess when set
+
+        with Pool(self.config["processes"], maxtasksperchild=1, initializer=init, initargs=(_abort,)) as pool:
+            # Using list comprehension to prevent new tasks from starting if an event occured
+            pool.map(extract_transform_load, [worker for worker in self.workers if not _abort.is_set()])
```

Comment:
I didn't notice any differences using a generator when testing locally. After additional testing it became apparent  conditional can be removed since the `extract_transform_load()` function will exit early if the abort event was set